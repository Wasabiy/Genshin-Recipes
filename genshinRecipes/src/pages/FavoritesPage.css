#displayBoxFav {
    margin: auto;
    padding: 1rem;
    text-align: center;
    width: fit-content;
    height: 1280px;
    overflow-y: scroll;
}

#recipesBoxFav {
    max-width: 700px;
    justify-content: space-between;
    align-content: center;
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    flex: 1;
    padding: 0;
    height: auto;
    font-size: small;
}
