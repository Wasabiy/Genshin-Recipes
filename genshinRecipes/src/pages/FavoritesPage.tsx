import {useState, useEffect} from "react";
import { Food, KeyFood, KeyIngredient } from "../models/interface.ts";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { reformatData } from "../utils/globalFunctions.ts";
import axios from 'axios'
import RecipeCard from "../components/RecipeCard.tsx";

function FavoritesPage() {
    const [favoriteRecipes, setFavoriteRecipes] = useState<KeyFood | null>();
    
    function fetchFavoriteRecipes() {
        const favoritedFoods: KeyFood[] = [];
        const data: any[] = [];
        for (let x = 0; x < localStorage.length; x++) {
            const key = localStorage.key(x);
            if (key && localStorage.getItem(key) == "true") {
                data.push(key);
                //data[key] = JSON.parse(localStorage.getItem(key) || '{}');
            }
        }
        console.log(data);

        axios.get('https://genshin.jmp.blue/consumables/food')
            .then((response1) => {
               const response:KeyFood[] = reformatData(response1.data, "KeyFood");
                for (const x in data) {
                    const selected = response.find(y => y.key == x);
                    if (selected) {
                        favoritedFoods.push(selected);
                    } 
                }
            });
    }
    
    useEffect(() => {
        fetchFavoriteRecipes();
        

        
    }, []);
    return (
        <>
            <h1>Favorites</h1>
        </>
    )
}
export default FavoritesPage;