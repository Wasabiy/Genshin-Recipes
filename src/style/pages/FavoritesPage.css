#displayBoxFav {
  margin: auto;
  padding: 1rem;
  text-align: center;
  width: fit-content;
}

#recipesBoxFav {
  max-width: 700px;
  justify-content: space-between;
  align-content: center;
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  flex: 1;
  padding: 0;
  height: auto;
  font-size: small;
}

#recipesBoxFav a {
  text-decoration: none;
  color: white;
}

#displayText {
  text-align: center;
}
