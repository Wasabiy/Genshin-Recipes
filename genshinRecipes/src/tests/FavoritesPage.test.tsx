import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { describe, expect, test, vi } from 'vitest';
import { MemoryRouter } from 'react-router-dom';
import FavoritesPage from '../pages/FavoritesPage.tsx';

vi.mock('../utils/globalFunctions', () => ({
  changeFavoriteState: vi.fn(),
  getFavoriteState: vi.fn(),
}));

const mockPush = vi.fn();

vi.mock('@hooks/useNavigation', () => {
  return {
    default: () => ({
      push: mockPush,
    }),
  };
});
describe('RecipesPage.tsx snapshot test', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <FavoritesPage />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
