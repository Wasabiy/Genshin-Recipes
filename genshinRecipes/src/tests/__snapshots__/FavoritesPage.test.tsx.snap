// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RecipesPage.tsx snapshot test > matches snapshot 1`] = `
<DocumentFragment>
  <section
    id="displayBoxFav"
  >
    <section
      id="recipesBoxFav"
    />
  </section>
</DocumentFragment>
`;
